src/encoded/commands/generate_ontology.py

```diff
@@ -1053,12 +1101,12 @@ def main():
         print('Processing: ', ontology['ontology_name'])
         if ontology.get('download_url', None) is not None:
             # get all the terms for an ontology
-            terms, v, deprecated = download_and_process_owl(ontology, connection, terms, deprecated, simple=args.simple)
+            terms, v, deprecated = download_and_process_owl(
+                    ontology, terms, deprecated, simple=args.simple, owlfile=args.owlfile)
             if not v and ontology.get('ontology_name').upper() == 'UBERON':
                 try:
-                    result = requests.get('http://svn.code.sf.net/p/obo/svn/uberon/releases/')
-                    release = result._content.decode('utf-8').split('</li>\n  <li>')[-1]
-                    v = release[release.index('>') + 1: release.index('</a>')].rstrip('/')
+                    result = requests.get('https://api.github.com/repos/obophenotype/uberon/git/refs/tags')
+                    v = result.json()[-1].get('ref').split('/v')[-1]
                 except Exception:
                     print('Unable to fetch Uberon version')
             if v and v != ontology.get('current_ontology_version', ''):
```

Comment:
I think that to respect the --full argument we want to change this line (1064) to:
```if (v and v != ontology.get('current_ontology_version', '')) or args.full:

Reply:
And also, to fix a bug I was running into (running localling anyways), the very next line after that should initialize `prev` to `[]`, i.e. ```prev = []```

Reply:
Absolutely agree on initializing prev.
In regard to the --full argument I had not considered the use case of reverting something to an earlier version of the ontology and I am not sure that if one tried to do so things would work exactly as hoped - which I think is the use case you are considering?  The --full option was intended as more of a checking mechanism where a file was generated that contained both terms to be posted or updated as well as existing unchanged terms that are usually filtered out so the file that was generated was 'complete' in reference to the db it was run against - the usual filtering reduces the file size and avoids unnecessary patches but sometimes it helps to see everything that exists in the ontology obtained from the maintainers if troubleshooting is required.  And generally in that case you don't need the ontology version info but I don't think it hurts.

Reply:
Okay yeah, I can take that out. I was just getting confused when generating an ontology file and wasn't getting any ontology (as opposed to term) records, so I figured the --full was intended to do that; but I didn't realize the fuller implications of that.